lib/name.mli

```diff
@@ -27,11 +27,20 @@ open Cstruct
 type label
 
 (** Domain name, as a list of strings. *)
-type domain_name = string list
+type domain_name
 
 (** Domain name map *)
 module Map: Map.S with type key = domain_name
 
+(** Create null domain name *)
+val empty: domain_name
```

Comment:
Oops, I meant in the documentation string "null" is a bit confusing. Does it mean using it is invalid? The word "empty" or "zero" or "root" is more descriptive, IMO.